scripts/RelayHubGasEstimation.ts

```diff
@@ -0,0 +1,641 @@
+import { UtilToken } from '@rsksmart/rif-relay-contracts'
+import { BigNumber, ContractReceipt, Wallet } from 'ethers'
+import {
+  RSK_URL,
+  createSmartWalletFactory,
+  createSupportedSmartWallet,
+  deployRelayHub,
+  getSuffixDataAndSignature,
+} from '../test/utils/TestUtils'
+import {
+  Penalizer,
+  RelayHub,
+  SmartWallet,
+  SmartWalletFactory,
+  TestRecipient,
+  TestVerifierEverythingAccepted,
+} from 'typechain-types'
+import { expect } from 'chai';
+
+const gasPrice = 1
+const gasLimit = 4e6
+const bgMagenta = '\x1B[45m';
+const underline = "\x1b[4m"
+const reset = '\x1b[0m';
+
+const cloneRelayRequest = (relayRequest: RelayRequest, override: {
+  request?: Partial<RelayRequestBody>;
+  relayData?: Partial<EnvelopingRequestData>;
+}) => {
+  return {
+    request: { ...relayRequest.request, ...override.request },
+    relayData: { ...relayRequest.relayData, ...override.relayData },
+  };
+};
+
+function logTitle(title: string) {
+  console.log(`${underline}${title}${reset}`);
+}
+
+function logGasOverhead(gasOverhead: BigNumber) {
+  console.log(`${bgMagenta}Enveloping Overhead Gas: ${gasOverhead.toString()}${reset}\n`);
+}
+
+const estimateGas = async () => {
+
+  const [
+    relayWorker,
+    relayManager,
+    relayOwner,
+    fundedAccount,
+    relayHubSigner,
+  ] = (await ethers.getSigners()) as [
+    SignerWithAddress,
+    SignerWithAddress,
+    SignerWithAddress,
+    SignerWithAddress,
+    SignerWithAddress,
+  ]
+
+  let penalizer: Penalizer;
+  let relayHub: RelayHub;
+  let verifier: TestVerifierEverythingAccepted;
+  let recipient: TestRecipient;
+  let token: UtilToken;
+  let owner: Wallet;
+  let forwarder: SmartWallet
+  let relayRequest: RelayRequest;
+
+  const deployAndSetup = async () => {
+    const provider = new ethers.providers.JsonRpcProvider(RSK_URL);
+  
+    const deployContract = <Contract>(contract: string) => {
+      return ethers
+        .getContractFactory(contract)
+        .then((contractFactory) => contractFactory.deploy() as Contract)
+    }
+  
+    owner = ethers.Wallet.createRandom().connect(provider)
+    penalizer = await deployContract<Penalizer>('Penalizer')
+    verifier = await deployContract<TestVerifierEverythingAccepted>(
+      'TestVerifierEverythingAccepted',
+    )
+    recipient = await deployContract<TestRecipient>('TestRecipient')
+    token = await deployContract<UtilToken>('UtilToken')
+    const smartWalletTemplate = await deployContract<SmartWallet>('SmartWallet')
+  
+    relayHub = await deployRelayHub(penalizer.address)
+  
+    await fundedAccount.sendTransaction({
+      to: owner.address,
+      value: ethers.utils.parseEther('1'),
+    })
+  
+    const factory = (await createSmartWalletFactory(
+      smartWalletTemplate,
+      false,
+      owner,
+    )) as SmartWalletFactory
+  
+    forwarder = (await createSupportedSmartWallet({
+      relayHub: relayHubSigner.address,
+      factory,
+      owner,
+      sender: relayHubSigner,
```

Comment:
Shouldn't the worker be the sender?

Reply:
I attempted this but it breaks. We would have to change the relayHub as well to be the worker.

Reply:
yes, you're right. It looks like it fails because we're sending the request directly to the SmartWalletFactory. Although here we're deploying a SW, this is not the real process, because the call should be performed against `RelayHub.deployCall`. I'm going to create a task for this to cover this specific scenario.